internal/moreos/moreos_func-e_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package moreos
+
+import (
+	"bufio"
+	"bytes"
+	"embed"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/func-e/internal/test/fakebinary"
+	"github.com/tetratelabs/func-e/internal/version"
+)
+
+var (
+	// fakeFuncESrc is a test source file used to simulate how func-e manages its child process
+	//go:embed testdata/fake_func-e.go
+	fakeFuncESrc []byte // Embedding the fakeFuncESrc is easier than file I/O and ensures it doesn't skew coverage
+
+	// Include the source imported by fakeFuncESrc directly and indirectly
+	// TODO: replace wildcard with {{goos}} after https://github.com/golang/go/issues/48348.
+	//go:embed moreos.go
+	//go:embed proc_*.go
+	moreosSrcDir embed.FS
+)
+
+// TestProcessGroupAttr_Kill tests sending SIGKILL to fake func-e.
+// On linux, we propagate SIGKILL to the child process as the configured SysProcAttr.Pdeathsig
+// in proc_linux.go.
+func TestProcessGroupAttr_Kill(t *testing.T) {
+	// This works only for linux, sending kill -9 on darwin will not kill the process, we need to kill
+	// via pgid or kill the child first.
+	if runtime.GOOS != "linux" {
+		t.Skip()
+	}
+	tempDir := t.TempDir()
+
+	// Build a fake envoy and pass the ENV hint so that fake func-e uses it
+	fakeEnvoy := filepath.Join(tempDir, "envoy"+Exe)
+	fakebinary.RequireFakeEnvoy(t, fakeEnvoy)
+	t.Setenv("ENVOY_PATH", fakeEnvoy)
+
+	fakeFuncE := filepath.Join(tempDir, "func-e"+Exe)
+	requireFakeFuncE(t, fakeFuncE)
+
+	stderr := new(bytes.Buffer)
+	stdout := new(bytes.Buffer)
+
+	// With an arg so fakeFuncE runs fakeEnvoy as its child and doesn't exit.
+	arg := string(version.LastKnownEnvoy)
+	cmd := exec.Command(fakeFuncE, "run", arg, "-c")
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	require.NoError(t, cmd.Start())
+
+	// Block until we reach an expected line or timeout.
+	reader := bufio.NewReader(stderr)
+	waitFor := "initializing epoch 0"
+	if !assert.Eventually(t, func() bool {
```

Comment:
any reason why this is using assert instead of require?
```suggestion
	if !require.Eventually(t, func() bool {
```

Reply:
Ah yeah, missed this one. Seems like a copy-and-paste error. Thanks!

Reply:
Updated. Thanks!